### ISSUE TITLE
Masked losses

### ISSUE THREAD

**Issue by markdjwilliams** [nerfstudio-project/nerfstudio#3475]
I may have identified an issue with how losses are computed when using masked images in the Splatfacto training method within Nerfstudio. It appears that the current implementation includes masked-out pixels in the loss calculations, which may artificially lower the loss values and affect the training process.

For L1 loss, the loss is currently computed as:

`Ll1 = torch.abs(gt_img * mask - pred_img * mask).mean()`

The `mean()` operation includes all pixels, including those that are masked out. This can result in an artificially low loss since irrelevant masked pixels still contribute to the denominator. When a mask is present (and not all zeroes) I suspect this should instead be:

`Ll1 = torch.abs(gt_img * mask - pred_img * mask) / torch.sum(mask)`

The SSIM loss suffers from a similar issue. There's a [comment](https://github.com/nerfstudio-project/nerfstudio/blob/67be06009eae045f25b1ce5a67c012cbaf57e2ab/nerfstudio/models/splatfacto.py#L708) in the code that perhaps alludes to this missing functionality.

While this might not significantly affect scenarios where masked-out areas are minimal, it could lead to inaccurate loss evaluations and potentially hinder the training process in cases with substantial masking.
